pkg/webhook/policy.go

```diff
@@ -173,7 +173,25 @@ func (h *validationHandler) Handle(ctx context.Context, req atypes.Request) atyp
 	if len(res) != 0 {
 		var msgs []string
 		for _, r := range res {
-			msgs = append(msgs, fmt.Sprintf("[denied by %s] %s", r.Constraint.GetName(), r.Msg))
+			// check for namespaceselector and sync resources
```

Comment:
This is the right idea, but may be moot if we can implement a rejection at the Rego level.

Reply:
Since we are rejecting across all namespaces, I think it's best to return an error to let users know they need to sync `Namespace` as part of the sync configuration.

Reply:
Correct. This error could also be returned via Rego, however. This would also have the advantage of not requiring the existence of the API Server to run the validation handler. Requiring the API Server means that this target could not be used with other enforcement points.

Finally, there is also the race condition to consider. This isn't just about whether the sync config is defined or not, but whether GK has received/processed the Namespace watch event.

Reply:
Sorry, part of my comment was inaccurate:

> This would also have the advantage of not requiring the existence of the API Server to run the validation handler. Requiring the API Server means that this target could not be used with other enforcement points.

It's good this is removed though, as the code would never have executed. By definition, any constraint with a namespace selector wouldn't show up in the validation results of a request that had no cached namespace.